File: Leetcode/3335 - Total Characters in String After Transformations I/solution.py
class Solution:
    def lengthAfterTransformations(self, s: str, t: int) -> int:
        MOD = 10**9 + 7
        char_count = [0] * 26

        for char in s:
            char_count[ord(char) - ord('a')] += 1

        for _ in range(t):
            temp = char_count[25]
            # Shift right from index 24 down to 0
            for i in range(25, 0, -1):
                char_count[i] = char_count[i - 1]
            char_count[0] = temp
            char_count[1] = (char_count[1] + temp) % MOD  # Apply mod here to keep values in range

        # Compute total sum under modulo
        return sum(count % MOD for count in char_count) % MOD

File: Leetcode/3335 - Total Characters in String After Transformations I/solution.ts
function lengthAfterTransformations(s: string, t: number): number {
    const MOD: number = 10 ** 9 + 7;
    const charCount: number[] = new Array<number>(26).fill(0);

    // Count initial character frequencies
    for (let i: number = 0; i < s.length; i++) {
        const index: number = s.charCodeAt(i) - 97; // ASCII of 'a' is 97
        charCount[index]++;
    }

    // Perform t transformations
    for (let i: number = 0; i < t; i++) {
        const temp: number = charCount[25];
        for (let j: number = 25; j > 0; j--) {
            charCount[j] = charCount[j - 1];
        }
        charCount[0] = temp;
        charCount[1] = (charCount[1] + temp) % MOD;
    }

    // Calculate total length modulo MOD
    const totalLength: number = charCount.reduce((sum: number, count: number) => (sum + count % MOD) % MOD, 0);

    return totalLength;
}